Procedural documentation for the join package (#4466)
influxdata/docs-v2: content/flux/v0.x/join-data/troubleshoot-joins.md

---

> title: Troubleshoot join operations

Reviewer: Thoughts on organizing "Issues" and "Solutions" in a table for "scan-ability"? Perhaps alphabetizing errors in a table? Recognize there's not a huge # of items, but may be nice?

---
title: Troubleshoot join operations
description: >
  Learn how to troubleshoot common behaviors and errors that may occur when using
  the [`join` package](/flux/v0.x/stdlib/join).
menu:
  flux_0_x:
    name: Troubleshoot joins
    parent: Join data
weight: 105
---

Learn how to troubleshoot common behaviors and errors that may occur when using
the [`join` package](/flux/v0.x/stdlib/join).

{{% note %}}
#### Submit issues for unexplained behaviors or errors

This is a "living" document that may be updated with common issues
that users may run into when using the [`join` package](/flux/v0.x/stdlib/join)
to join data. If you have questions about a behavior or error that is not
documented here, please submit an issue on either the InfluxData Documentation
or Flux GitHub repositories:

- [Submit a documentation issue](https://github.com/influxdata/docs-v2/issues/new/choose)
- [Submit a Flux issue](https://github.com/influxdata/flux/issues/new/choose)
{{% /note %}}

> - [Troubleshoot join behaviors](#troubleshoot-join-behaviors)

Reviewer: understand why you included, but may want to remove for simplicity until we have more behaviors?

  - [Columns explicitly mapped in the join are null](#columns-explicitly-mapped-in-the-join-are-null)
- [Troubleshoot join error messages](#troubleshoot-join-error-messages)

## Troubleshoot join behaviors

### Columns explicitly mapped in the join are null

> In some cases, the output of your join operation may include _null_ values in
> columns you would expect to return with non-null values. This is likely caused
> by one of the following problems:

Reviewer: ```suggestion
In some cases, your join output may include _null_ values in
columns where you expect non-null values. This may be caused by one of the following issues:
```

- **The group keys of your input streams are different**

>   **Problem**:

Reviewer: Problem >> Issue

Author: 👍🏻

>   The group keys of each input stream needs to be the same when using the `join`

Reviewer: ```suggestion
  The group keys of each input stream aren't the same
```

>   package to join data. Functions in the `join` package use group keys to

Reviewer: ```suggestion
Functions in the `join` package use group keys to
```
